woof-code/support/petbuilds.sh

```diff
@@ -68,7 +68,8 @@ for NAME in $PETBUILDS; do
         continue
     fi
 
-    HASH=`cat ../DISTRO_PKGS_SPECS-${DISTRO_BINARY_COMPAT}-${DISTRO_COMPAT_VERSION} ../DISTRO_COMPAT_REPOS ../DISTRO_COMPAT_REPOS-${DISTRO_BINARY_COMPAT}-${DISTRO_COMPAT_VERSION} ../DISTRO_PET_REPOS ../rootfs-petbuilds/${NAME}/petbuild 2>/dev/null | md5sum | awk '{print $1}'`
+    tar -czf /tmp/petbuild.tar.gz ../rootfs-petbuilds/${NAME}/ # /tmp/petbuild.tar.gz is also used for generating HASH; can be used to compare any change in any file/folder of the petbuild directory
```

Comment:
The output of `tar` is no-deterministic, it contains things like timestamps and the order of files can change. The hash is almost guaranteed to change every time this command runs, so you're essentially disabling the cache completely.

Reply:
Thanks, I didn't know about that.

We will need some other approach then. Closing this PR.

Reply:
@dimkr, I have another idea.

How about using the output of `diff -r [CACHED_PETBUILD_DIR] [WOOF-CE_PETBUILD_DIR]`? If the contents are different, `diff` will output something; otherwise, its output will be empty.

Will that work?

Reply:
I don't know if it's done in deterministic order, and I think the output contains timestamps.

Reply:
I'll test this idea tomorrow and open a PR if it works.

Reply:
If you don't hash it, you need to store it on disk as-is, increasing cache size and making the comparison that determines whether or not rebuild is needed slower.